### DISCUSSION TITLE
Train and Validation DataLoaders

### DISCUSSION THREAD

**Discussion by AndreyBocharnikov** [NVIDIA-NeMo/NeMo#2933] [Q&A]
Hello! I am quite new to NeMo and pytorch lightning, so I am a bit confused about calling `trainer.fit(model)` without `train_dataloader` and `validation_dataloader` as it is done in many examples e.g. `examples/tts/talknet_spect.py`, but I fount that `self._train_dl` and `self._validation_dl` are being initialised as train and validation dataloaders respectively in line 204 and 207 of `nemo/collections/tts/models/talknet.py` script, but I can not find where they are being used? Is it pytorch lightning convention that such fields are used as data loaders or they they are used somehow in NeMo? Couldn't find either of options. Can somebody please help me understand this, thank you.

**Comment by VahidooX** [comment]
These fields are used by nemo to keep the track of the data loaders and return them to the PTL. If you search the whole repo, you will find them. Some examples are inside nemo/classes/ModelPT.py file. The data loaders are created automatically when you create a model by passing the config as it contains the data loaders' info. No need to create them separately. PTL would use these data loaders to create batches and handle the whole process of the training.
